lib/modules/manager/npm/update/dependency/index.ts

```diff
@@ -25,7 +26,17 @@ function replaceAsString(
   }
   // Look for the old version number
   const searchString = `"${oldValue}"`;
-  const newString = `"${newValue}"`;
+  let newString = `"${newValue}"`;
+
+  const escapedDepName = escapeRegExp(depName);
+  const patchRe = regEx(`^(patch:${escapedDepName}@(npm:)?).*#`);
+  const match = patchRe.exec(oldValue);
+  if (match) {
+    const patch = oldValue.replace(patchRe, `${match[1]}${newValue}#`);
```

Comment:
```suggestion
    const patch = oldValue.replace(match[0], `${match[1]}${newValue}#`);
```
can't we simply replace our match instead of executing the regEx again?

Reply:
I'll commit and we see :)